### ISSUE TITLE
Enhancement needed for vomsxrd packaging

### ISSUE THREAD

**Issue by simonmichal** [xrootd/xrootd#1230]
As summarised by @olifre in email conversation:

1. The problem Fabrizio hit: Existing packages with 
    Requires: vomsxrd > 1:0.2.0
  or similar are not satisfied by:
    Provides:  vomsxrd
  as xrootd.spec contains now. 
  So likely, that needs to be changed to either:
    Provides:  vomsxrd = %{epoch}:%{version}-%{release}
  or maybe even:
    Provides:  vomsxrd = 1:0.8.0
  since the last vomsxrd release which is the predecessor of xrootd-voms was 0.8.0,
  and any package requiring something newer should change to require xrootd-voms. 

2. The problem Michal identified, which is a bump in the migration path. 
  When you have vomsxrd installed and do yum update / install xrootd-voms, this happens:
     file /usr/lib64/libXrdSecgsiVOMS-4.so from install of xrootd-libs-1:4.12.3-1.el7.x86_64 conflicts with file from package vomsxrd-1:0.3.0-1.el7.cern.x86_64
     file /usr/lib64/libXrdSecgsiVOMS-4.so from install of xrootd-voms-1:4.12.3-1.el7.x86_64 conflicts with file from package vomsxrd-1:0.3.0-1.el7.cern.x86_64
  So what Fabrizio and I (and likely others) did was "yum remove vomsxrd", but indeed this might be done more smoothly,
  and "Obsoletes:" is probably the right thing as Michal pointed out. 
  However, I think we need _both_ "Provides:" (to satisfy existing packages depending on vomsxrd, see case (1)) and "Obsoletes:". 

3. The problems Wei pointed out. 
  Referring to "libXrdSecgsiVOMS-4.so exists in xrootd-libs-4.12.1-1 @ xrootd.stable repo. It works with the ATLAS VO but not with Dteam VO.":
   We use 4.12.3 in Bonn and the issue is fixed for us, both dteam and wlcg VO work. 
   We also saw this issue with 4.12.1, but with 4.12.2 it was fixed. For us, this was caused by:
    https://github.com/xrootd/xrootd/issues/1196
   which indirectly caused authentication issues for some VOs (not due to the VOs, actually, but due to the argument parsing issues causing a messed-up VOMS library configuration). 

So in short, (3) is already fixed (at least in my tests), fixing (2) would make the migration path smoother so you don't have to manually remove the old vomsxrd,
and fixing (1) is needed to satisfy existing consumers of vomsxrd (e.g. dmlite.dpm).

**Comment by simonmichal** [comment]
I pushed a fix for 1) and 2) : 398ed1e8404d28d1bc979f7babc5317b3b5a2a81, plus I back-ported 2fefd73fe6eb3caa2f7e73560a85b6fbd92a07ce

A release 4.12.4 release candidate should be available by the end of the week for testing.
